langchain/vectorstores/pinecone.ts

```diff
@@ -61,7 +61,7 @@ export class PineconeStore extends VectorStore {
 
     const result: [Document, number][] = [];
 
-    for (const res of results.matches) {
+    for (const res of results.data.matches) {
```

Comment:
this is the main change, the official client return the match inside `results.data.matches` instead of `results.matches`.